packages/flutter/lib/src/widgets/scroll_physics.dart

```diff
@@ -57,6 +57,39 @@ class ScrollPhysics {
   /// (e.g. [shouldAcceptUserOffset]) will defer to [AlwaysScrollableScrollPhysics].
   final ScrollPhysics parent;
 
+  /// Provides a heuristic to determine if expensive frame-bound tasks should be
+  /// deferred.
+  ///
+  /// The velocity parameter must not be null, but may be positive, negative, or
+  /// zero.
+  ///
+  /// This can be used to determine whether decoding or fetching complex data
+  /// for the currently visible part of the viewport should be delayed
+  /// to avoid doing work that will not have a chance to appear before a new
+  /// frame is rendered.
+  ///
+  /// For example, a list of images would use this logic to delay decoding
+  /// images until scrolling is slow enough to actually render the decoded
+  /// image to the screen.
+  ///
+  /// The default implementation is a heuristic that compares the current
+  /// scroll velocity in local logical pixels to the longest side of the window
+  /// in physical pixels. Implementers can change this heuristic by overriding
+  /// this method and providing their custom physics to the scrollable widget.
+  /// For example, an application that changes the local coordinate system with
+  /// a large perspective transform could provide a more or less aggressive
+  /// heuristic depending on whether the transform was increasing or decreasing
+  /// the overall scale between the global screen and local scrollable
+  /// coordinate systems.
+  bool recommendDeferredLoading(double velocity) {
```

Comment:
seems like at a minimum we should pass in a Context here

Reply:
or if that's not possible, the ScrollMetrics, or some such

Reply:
looks like you could pass both in

Reply:
You mean a BuildContext?  

Yes, I could pass both in.  I can see how ScrollMetrics could be helpful, I'm less clear on how a BuildContext would be.

Reply:
Went with the build context of the original calling element - per chat discussion could be useful for getting MediaQuery or like